processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -153,19 +202,21 @@ public void close() throws IOException
     private final BufferAggregator[] aggregators;
     private final PositionMaintainer positionMaintainer;
 
-    private final Map<ByteBuffer, Integer> positions = Maps.newTreeMap();
+    private final Map<ByteBuffer, Integer> positions;
     // GroupBy queries tend to do a lot of reads from this. We co-store a hash map to make those reads go faster.
     private final Map<ByteBuffer, Integer> positionsHash = Maps.newHashMap();
 
     public RowUpdater(
         ByteBuffer metricValues,
         BufferAggregator[] aggregators,
-        PositionMaintainer positionMaintainer
+        PositionMaintainer positionMaintainer,
+        List<QueryDimensionInfo> dimInfo
     )
     {
       this.metricValues = metricValues;
       this.aggregators = aggregators;
       this.positionMaintainer = positionMaintainer;
+      this.positions = Maps.newTreeMap(makeKeyComparator(dimInfo));
```

Comment:
TreeMap was used in the original GroupByEngine code being moved, my goal with this PR was to create the interface and move type-specific code but not change functionality generally/target performance optimizations